versionedController/v1/configuration/configuration.go

```diff
@@ -53,79 +57,82 @@ func (configurationController *Controller) Put(c *gin.Context) {
 		c.JSON(http.StatusUnauthorized, err.Error())
 		return
 	}
-
-	err = c.BindJSON(configurationModel)
+	// 3. Validate request body
+	configModel := &Model{}
+	err = c.BindJSON(configModel)
 	if err != nil {
 		log.Error(err)
 		c.JSON(http.StatusNotAcceptable, gin.H{
 			"message": "Unable to parse JSON",
 		})
 		return
 	}
+	// 4. Persist the configuration for further usage
+	configurationController.updateToK8s(c, configModel)
+}
 
-	if configurationModel.ClientID != nil && configurationModel.EnableGithub != nil {
-		credentialType := c.Query("type")
-		var data map[string]string
-
-		switch credentialType {
-		case string(models.GithubAuth):
-			{
-				data = map[string]string{
-					types.GITHUB_CLIENT_ID:     *configurationModel.ClientID,
-					types.GITHUB_CLIENT_SECRET: *configurationModel.ClientSecret,
-				}
-			}
-		default:
-			{
-				c.Writer.WriteHeader(http.StatusBadRequest)
-				return
-			}
-		}
-
-		secret, err := k8s.ClientSet.CoreV1().Secrets(types.DefaultNamespace).Get(c.Request.Context(), types.Credentials, metav1.GetOptions{})
-		if err != nil {
-			c.String(http.StatusInternalServerError, "Error storing credentials : %v", err)
-			log.Errorln("Error getting secret ", err)
-			return
-		}
-
-		secret.StringData = data
-		_, err = k8s.ClientSet.CoreV1().Secrets(types.DefaultNamespace).Update(c.Request.Context(), secret, metav1.UpdateOptions{})
-		if err != nil {
-			c.String(http.StatusInternalServerError, "Error storing credentials : %v", err)
-			log.Errorln("Error updating secret ", err)
-			return
-		}
-
-		controller.Server.GithubConfig.ClientID = *configurationModel.ClientID
-		controller.Server.GithubConfig.ClientSecret = *configurationModel.ClientSecret
+// updateToK8s updates the value of Oauth secrets in secrets & configmaps
+// TODO: Need to lock this operation, so only one request can update it at a time
+// TODO(vharsh): Use pflags to configure all of these
+func (configurationController *Controller) updateToK8s(c *gin.Context, requestModel *Model) {
```

Comment:
Where is this used? If admin makes any changes to the oauth configuration?

Reply:
Yes, this is the function which takes the request model and updates it to the configmap.